pages/writing.md

```diff
@@ -45,23 +45,131 @@ juliaup status
 
 ## Development environments
 
+Depending on your experience and how you use Julia, some ways of writing code may be more familiar or useful to you than others.
+Data scientists commonly prefer notebook environments such as those provided by [Jupyter](https://jupyter.org/) through the [IJulia.jl](https://github.com/JuliaLang/IJulia.jl) package, or Julia's own reactive notebooks from [Pluto.jl](https://plutojl.org/).
+On the other hand, package developers or those who write a lot of scripts will prefer the more traditional programming environment provided by a text editor or integrated development environment (IDE).
+
+For some, the Julia REPL (Read Evaluate Print Loop) itself is all that's necessary.
+The REPL's primary function is to run code in "Julian" mode, but it also has a number of other modes that expand what can be done from inside Julia.
+Each mode is enterred by typing a character into the REPL from Julian mode, and can be exited by deleting this character with backspace.
+
+### Help mode (`?`)
+By entering a `?`, you can query information and metadata about Julia objects and unicode symbols simply by typing their name into the command line.
+For functions, types, and variables, the query fetches things such as documentation, type fields and supertypes, and in which file the object is defined.
+<!-- How do you do syntax highlighting for the Julia REPL? -->
+```
+help?> Int
```

Comment:
it would be nice for this to be run by Franklin itself

Reply:
It would!

Reply:
https://github.com/tlienart/Franklin.jl/issues/818